### PR TITLE
Respect block gas limit

### LINKED ISSUE #2699: Respect block gas limit when sending batch transfer
We have to limit # of recipients in batchTransfer so the total gas cost won't exceed gas block limit.

### PR DESCRIPTION
**Pull request by Krigpl**
So we don't try to send a transaction with higher gas limit then the block gas limit - it wouldn't get accepted to the pool at all.

Fixes #2699

### CODE REVIEW

File: golem/ethereum/paymentprocessor.py [hunk 1]
```diff
@@ -194,13 +196,24 @@ def __get_next_batch(
         eth_balance = eth_balance - self.ETH_BATCH_PAYMENT_BASE
         ind = 0
         eth_per_payment = self.get_gas_cost_per_payment()
+        gas_limit = \
+            self._sci.get_latest_block().gas_limit * self.BLOCK_GAS_LIMIT_RATIO
+        payees = set()
```
Comment (mfranciszkiewicz) [inline golem/ethereum/paymentprocessor.py:201]: Are payments per payee summed?

Comment (Krigpl) [reply golem/ethereum/paymentprocessor.py:201]: Yes. Inside `sci.batch_transfer`

**Review by mfranciszkiewicz — APPROVED** [review]

File: golem/ethereum/paymentprocessor.py [hunk 2]
```diff
@@ -194,13 +196,24 @@ def __get_next_batch(
...
         for p in payments:
             if p.processed_ts > closure_time:
                 break
             gntb_balance -= p.value
-            eth_balance -= eth_per_payment
```
Comment (maaktweluit) [inline golem/ethereum/paymentprocessor.py:201]: no more need for updating the ETH balance?

**Review by maaktweluit — APPROVED** [review]
LGTM!

One small q, reply can be enough

File: golem/ethereum/paymentprocessor.py (hunk 2 above)
Comment (Krigpl) [reply golem/ethereum/paymentprocessor.py:201]: Yeah, previous estimation wasn't accounting for multiple payments to a single provider, current version does so the check is performed in a different way.

**Pull request merged**